### PR TITLE
Update and reduce Maven dependencies in Gradle distribution

### PR DESCRIPTION
**Pull request by jjohannes**
This:
- Removes all code in `maven` subproject that is no longer needed due to the removal of the old `maven` plugin.
- Upgrades to latest Maven (3.6.3)
- Reduces the Maven dependencies of the `dependency-management` and `maven`  subprojects to a minimum.
   - We only need the ability to deal with settings of local Maven repo and the functionality to build a model of a POM when generating Metadata
- Changes the maven2gradle functionality of `gradle init` to download dependencies on demand from mavenCentral()
   - Without this adjustment, the Maven upgrade would cause the distribution to grow by ~2mb and a few additional dependencies

With the reduction, we remove 20 jars (~2mb) from the bin distribution despite the upgrade.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -16,7 +16,7 @@ org.gradle.vfs.verbose=true
 org.gradle.vfs.debug=true
 
 # If you're experimenting with changes and don't want to update the verification file right away, please change the mode to "lenient" (not "off")
-org.gradle.dependency.verification=strict
+org.gradle.dependency.verification=lenient
```
Comment (melix) [inline gradle.properties:19]: :x: Don't forget to remove this before merging

File: subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/BuildInitPlugin.java [hunk 2]
```diff
@@ -48,6 +49,9 @@ public void apply(Project project) {
 
                 initBuild.onlyIf(new InitBuildOnlyIfSpec(buildFile, hasSubProjects, project.getLayout(), initBuild.getLogger()));
                 initBuild.dependsOn(new InitBuildDependsOnCallable(buildFile, hasSubProjects, project.getLayout()));
+
+                ProjectInternal.DetachedResolver detachedResolver = ((ProjectInternal) project).newDetachedResolver();
```
Comment (melix) [inline subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/BuildInitPlugin.java:53]: :+1:

File: subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/maven/PomProjectInitDescriptor.java [hunk 3]
```diff
@@ -73,16 +82,35 @@ public String getSourceBuildDescription() {
         return "Maven";
     }
 
+    @Override
+    public void configureClasspath(ProjectInternal.DetachedResolver detachedResolver, ObjectFactory objects) {
+        DependencyHandler dependencies = detachedResolver.getDependencies();
+        Configuration config = detachedResolver.getConfigurations().detachedConfiguration(
+            dependencies.create("org.apache.maven:maven-core:3.6.3"),
```
Comment (melix) [inline subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/maven/PomProjectInitDescriptor.java:89]: Would be nice to be avoid having those dependencies hardcoded in the sources, but that's good for now.

File: subprojects/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy [hunk 4]
```diff
@@ -37,7 +37,7 @@ import static org.hamcrest.MatcherAssert.assertThat
 
 abstract class DistributionIntegrationSpec extends AbstractIntegrationSpec {
 
-    protected static final THIRD_PARTY_LIB_COUNT = 144
+    protected static final THIRD_PARTY_LIB_COUNT = 124
```
Comment (melix) [inline subprojects/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy:40]: :tada:

**Review by melix — APPROVED** [review]
Very nice!

File: subprojects/distributions-dependencies/build.gradle.kts [hunk 5]
```diff
@@ -141,7 +119,7 @@ dependencies {
         api(libs.cglib)                 { version { strictly("3.2.6") }}
         api(libs.equalsverifier)        { version { strictly("2.1.6") }}
         api(libs.flightrecorder)        { version { strictly("7.0.0-alpha01") }}
-        api(libs.guice)                 { version { strictly("2.0") }}
+        api(libs.guice)                 { version { require("2.0") }}
```
Comment (ljacomet) [inline subprojects/distributions-dependencies/build.gradle.kts:122]: Is this a leftover from testing?

File: subprojects/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy (hunk 4 above)
Comment (ljacomet) [inline subprojects/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy:40]: Woot 🎉 !

**Review by ljacomet — APPROVED** [review]
Excellent!

File: gradle.properties (hunk 1 above)
Comment (jjohannes) [reply gradle.properties:19]: Yes. I still have to update and check the verification metadata (TODO is in the description). Just wanted to make sure we want to go with this before doing it.

File: subprojects/distributions-dependencies/build.gradle.kts (hunk 5 above)
Comment (jjohannes) [reply subprojects/distributions-dependencies/build.gradle.kts:122]: No. This is a test-only dependency and running tests in `build-init` actually needs a higher version now. But this is also used in another project (only for testing). But I can check if that other project can be upgraded to newer Guice. Then I can use strictly with the higher Guice version here an it is less confusing.

But I wonder if this global `strictly`s here for test-only dependencies should go. They don't really make sense, right?

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by lptr** [comment]
Looks like this PR introduced an `org.gradleinternal` (sic!) package. This was unintentional, right?

**Comment by jjohannes** [comment]
@lptr this is a long time ago and apparently this is not documented here, but I _think_ this was on purpose, because of the `org.gradleinternal...Maven2GradleWorkAction`. I think `org.gradle.*` code is put in some other scope in the workers and that action should be treated like "user code". But this is very vague memory. I think if it is an issue, some test will fail if it's changed.

**Pull request merged**